### PR TITLE
remove the excess locks

### PR DESCRIPTION
**Pull request by inabao**

### CODE REVIEW

File: src/algorithm/hnswlib/hnswalg.cpp [hunk 1]
```diff
@@ -932,8 +930,6 @@ HierarchicalNSW::DeserializeImpl(StreamReader& reader, SpaceInterface* s, size_t
 
 const float*
 HierarchicalNSW::getDataByLabel(LabelType label) const {
-    std::lock_guard<std::mutex> lock_label(getLabelOpMutex(label));
-
     std::unique_lock<std::mutex> lock_table(label_lookup_lock_);
```
Comment (inabao) [inline src/algorithm/hnswlib/hnswalg.cpp:933]: The label_lookup_lock_ ensures safety, so no additional locks are needed.

Comment (jiaweizone) [reply src/algorithm/hnswlib/hnswalg.cpp:933]: L939 unlocked the global lock object `lock_table` then L941 operation is not thread-safe ?

File: src/algorithm/hnswlib/hnswalg.cpp [hunk 2]
```diff
@@ -954,8 +950,6 @@ HierarchicalNSW::getDataByLabel(LabelType label) const {
 void
 HierarchicalNSW::markDelete(LabelType label) {
     // lock all operations with element by label
-    std::unique_lock<std::mutex> lock_label(getLabelOpMutex(label));
```
Comment (jiaweizone) [inline src/algorithm/hnswlib/hnswalg.cpp:957]: ditto, L961 is not thread-safe ?

Comment (inabao) [reply src/algorithm/hnswlib/hnswalg.cpp:957]: I have removed the unlock.

File: src/algorithm/hnswlib/hnswalg.cpp (hunk 1 above)
Comment (inabao) [reply src/algorithm/hnswlib/hnswalg.cpp:933]: I have removed the unlock.

**Review by LHT129 — APPROVED** [review]
LGTM

**Review by wxyucs — APPROVED** [review]
lgtm

**Pull request merged**